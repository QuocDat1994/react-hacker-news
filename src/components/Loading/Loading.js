import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSpinner } from "@fortawesome/free-solid-svg-icons";

import "./Loading.scss";

const Loading = () => {
  return (
    <div className="loading">
      <FontAwesomeIcon icon={faSpinner} spin />
      <div>Now Loading...</div>
    </div>
  );
};

export default Loading;
