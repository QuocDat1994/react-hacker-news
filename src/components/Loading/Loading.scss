@import "../../styles/mixin";
@import "../../styles/color";

.loading {
  @include flex-center;
  flex-direction: column;

  font-size: 40px;
  color: $primary-color;
  min-height: 500px;

  &.loading--small {
    font-size: 30px;

    .loading__text {
      display: none;
    }
  }
}
