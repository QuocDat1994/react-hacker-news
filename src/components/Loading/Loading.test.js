import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Loading from "./Loading";

Enzyme.configure({ adapter: new Adapter() });

describe("Loading", () => {
  it("should render without error", () => {
    const wrapper = shallow(<Loading />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render with props = false", () => {
    const wrapper = shallow(<Loading isLoadingMore={false} />);
    expect(wrapper.find(".loading").length).toBe(1);
  });

  it("should render with props = true", () => {
    const wrapper = shallow(<Loading isLoadingMore={true} />);
    expect(wrapper.find(".loading--small").length).toBe(1);
  });
});
