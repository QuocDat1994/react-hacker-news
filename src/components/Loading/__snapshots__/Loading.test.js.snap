// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render without error 1`] = `ShallowWrapper {}`;
